src/plot_api/helpers.js

```diff
@@ -490,6 +490,9 @@ exports.coerceTraceIndices = function(gd, traceIndices) {
     else if(!Array.isArray(traceIndices) || !traceIndices.length) {
         return gd.data.map(function(_, i) { return i; });
     }
+    else if(Array.isArray(traceIndices)) {
+        return traceIndices.filter(function(i) {return i < gd.data.length;});
+    }
```

Comment:
Should I do a `Lib.warn('indice is not number or is out of bound')?

Reply:
Hooray for `lib.isIndex`. I really should spend some time to find out all the functions we have in `lib`!